src/main/java/org/opensearch/sdk/ExtensionSettings.java

```diff
@@ -96,12 +100,26 @@ public String toString() {
      * @throws IOException if there is an error reading the file.
      */
     public static ExtensionSettings readSettingsFromYaml(String extensionSettingsPath) throws IOException {
+        Yaml yaml = new Yaml(new SafeConstructor());
         URL resource = Extension.class.getResource(extensionSettingsPath);
         if (resource == null) {
-            return null;
+            throw new IOException("extension.yml does not exist at path [" + extensionSettingsPath + "]");
+        }
+        try (InputStream inputStream = Files.newInputStream(Path.of(resource.toURI()))) {
+            Map<String, Object> extensionMap = yaml.load(inputStream);
+            inputStream.close();
```

Comment:
This should be `flush()`.   

The `close()` method does not automatically flush (in fact the Javadocs say it [does nothing](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html#close())), and you're using try-with-resources so it will be "closed" automatically at the end of the try block.